test/ScanDependencies/package_interface.swift

```diff
@@ -0,0 +1,30 @@
+// RUN: %empty-directory(%t)
+// RUN: split-file %s %t
+
+// RUN: %target-swift-frontend -emit-module %t/Bar.swift -I %t \
+// RUN:   -module-name Bar -package-name barpkg \
+// RUN:   -enable-library-evolution -swift-version 5 \
+// RUN:   -emit-module-interface-path %t/Bar.swiftinterface \
+// RUN:   -emit-private-module-interface-path %t/Bar.private.swiftinterface \
+// RUN:   -emit-package-module-interface-path %t/Bar.package.swiftinterface
+
+// RUN: %target-swift-frontend -scan-dependencies -o %t/deps.json -I %t -experimental-package-interface-load \
+// RUN:   %t/Client.swift -module-name Client -package-name barpkg -swift-version 5
```

Comment:
Can we add a test that ensures we use the `.private.` interface if the scanning invocation does not specify `-package-name`? If that is the expected behavior.

Reply:
Done. It does require the experimental flag that took me a while to figure out.